### PR TITLE
feat(environments): Add environment list to project settings

### PR DESCRIPTION
**Pull request by lynnagara**

### CODE REVIEW

**Comment by getsentry-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">You should update CHANGES due to the size of this PR</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">danger</a>
</p>

File: src/sentry/static/sentry/app/views/projectEnvironments.jsx [hunk 1]
```diff
@@ -0,0 +1,64 @@
+import React from 'react';
+import Reflux from 'reflux';
+import createReactClass from 'create-react-class';
+
+import EnvironmentStore from '../stores/environmentStore.jsx';
```
Comment (billyvg) [inline src/sentry/static/sentry/app/views/projectEnvironments.jsx]: extra file ext

File: src/sentry/static/sentry/app/views/projectEnvironments.jsx [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+import Panel from './settings/components/panel';
+import PanelHeader from './settings/components/panelHeader';
+import PanelBody from './settings/components/panelBody';
+import EmptyMessage from './settings/components/emptyMessage';
+import {t} from '../locale';
+
+import Row from './settings/components/row';
```
Comment (billyvg) [inline src/sentry/static/sentry/app/views/projectEnvironments.jsx]: Let's use `<PanelItem>` - I'll have a PR to remove `Row` (they are dupes)

File: src/sentry/static/sentry/app/views/projectEnvironments.jsx [hunk 3]
```diff
@@ -0,0 +1,64 @@
+import React from 'react';
+import Reflux from 'reflux';
+import createReactClass from 'create-react-class';
+
+import EnvironmentStore from '../stores/environmentStore.jsx';
+
```
Comment (billyvg) [inline src/sentry/static/sentry/app/views/projectEnvironments.jsx]: nit: remove \n

File: src/sentry/static/sentry/app/views/projectEnvironments.jsx [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+import Panel from './settings/components/panel';
+import PanelHeader from './settings/components/panelHeader';
+import PanelBody from './settings/components/panelBody';
+import EmptyMessage from './settings/components/emptyMessage';
+import {t} from '../locale';
+
```
Comment (billyvg) [inline src/sentry/static/sentry/app/views/projectEnvironments.jsx]: nit: remove \n

File: src/sentry/static/sentry/app/views/projectEnvironments.jsx [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+import Row from './settings/components/row';
+import SettingsPageHeader from './settings/components/settingsPageHeader';
+import ListLink from '../components/listLink';
+
+const ProjectEnvironments = createReactClass({
+  mixins: [Reflux.listenTo(EnvironmentStore, 'onEnvironmentsChange')],
```
Comment (billyvg) [inline src/sentry/static/sentry/app/views/projectEnvironments.jsx]: Do we use `EnvironmentStore` elsewhere? If so, a HoC would be nice, especially with planned redux change.

**Review by billyvg — APPROVED** [review]

File: src/sentry/static/sentry/app/views/projectEnvironments.jsx (hunk 5 above)
Comment (lynnagara) [reply src/sentry/static/sentry/app/views/projectEnvironments.jsx]: 👍 Will open a PR for this

**Pull request merged**